src/dolphin/timeseries.py

```diff
@@ -607,28 +607,33 @@ def correlation_to_variance(correlation: ArrayLike, nlooks: int) -> Array:
 
 def select_reference_point(
     ccl_file_list: Sequence[PathOrStr],
-    amp_dispersion_file: PathOrStr,
+    condition_file: PathOrStr,
     output_dir: Path,
+    condition_func: Callable = np.argmin,
```

Comment:
We could annotate the expected input types are return value of the callable as well

```suggestion
    condition_func: Callable[[ArrayLike], int] = np.argmin,
```

or (see the above comment)

```suggestion
    condition_func: Callable[[ArrayLike], tuple[int, ...]] = np.argmin,
```